### DISCUSSION TITLE
get image bytes as one byte per pixel

### DISCUSSION THREAD

**Discussion by yymsc** [ImageMagick/ImageMagick#5576] [Digital Image Processing] [answered]
Hi,
I'm really quite new to ImageMagick and please apologize if this is a stupid request ;-)
Using the MagickWand Image API for C I need to get the pixels from a black/white TIFF image as an array of one byte per pixel.
Eg. let's say the first pixel (top left) is black and the next two pixels are white followed by one more black pixel I would like to get the following array: FF 00 00 FF
Is this possible? I tried using MagickSetImageDepth, MagickSetImageColorspace and some other functions, but no luck.
Any help appreciated.

thx,
Matt

**Comment by snibgo** [comment, marked as answer]
There are sample MagickWand programs at https://imagemagick.org/script/magick-wand.php . Sadly, they haven't been completely upgraded for IM v7. For example, in contrast.c, `MagickPixelPacket pixel;` needs to be `PixelInfo pixel;` and `PixelSetMagickColor(...)` needs to be `PixelSetPixelColor(...)`.

But that program shows how you can read pixel values, on a nominal scale of 0 to QuantumRange. As you need a nominal scale of 0 to 255, then (obviously) multiply values by 255.0/QuantumRange.

**Reply by urban-warrior** [reply]
Updated the docs/source per your concerns.
